### ISSUE TITLE
Make User Agent reflect GeckoView in `gecko` configuration

### ISSUE THREAD

**Issue by ncalexan** [mozilla-mobile/firefox-tv#1879]
## Steps to reproduce

In `system` and `gecko` configurations, observe `navigator.userAgent` in content.

### Expected behavior

`navigator.userAgent` should vary between `system` and `gecko` configurations.

### Actual behavior

`navigator.userAgent` is constant across `system` and `gecko` configurations.

### Device information
* Fire TV device: Cube, but it doesn't matter -- this is hard-coded.
* Latest affected Firefox version: trunk.
* Earliest affected Firefox version: probably every one since the `gecko` configuration was landed.

**Comment by ncalexan** [comment]
My use case is running [browsertime](https://github.com/sitespeedio/browsertime) against Firefox TV, and I can't distinguish my system WebView runs from my GeckoView runs.

**Comment by devinreams** [comment]
@cpeterso @ncalexan we couldn't quite tell, does this block your progress and development?

**Comment by ncalexan** [comment]
> @cpeterso @ncalexan we couldn't quite tell, does this block your progress and development?

No.  I've applied a trivial patch locally and I have other ways to determine this.  The logic is non-trivial (very!) but very helpful so I'd like it to get done for the GeckoView version sooner rather than later.

**Comment by cpeterso** [comment]
mozilla-mobile/android-components#1018 just landed a fix so A-C will read GV's default User-Agent string. Would updating FFTV's A-C version fix this issue?

**Comment by ncalexan** [comment]
> [mozilla-mobile/android-components#1018](https://github.com/mozilla-mobile/android-components/issues/1018) just landed a fix so A-C will read GV's default User-Agent string. Would updating FFTV's A-C version fix this issue?

Not immediately because FFTV does pretty wild things: https://github.com/mozilla-mobile/firefox-tv/blob/f2d1ab1bacc7dae1339da5a8479cea1eb48384bd/app/src/gecko/java/org/mozilla/tv/firefox/webrender/WebRenderComponents.kt#L23.  Stop doing that (and passing it through) and FFTV will take GV's string, which may or may not be the desired outcome.

**Comment by ncalexan** [comment]
@athomasmoz and I talked about this, and conclude that we're not going to do this until we can make Firefox TV's User Agent agree with a more mainline browser's User Agent.  That is, we might make this match Firefox for Android (or a successor) but we won't risk User Agent-based blocking in the interim.
